### ISSUE TITLE
is there a way to get rid of cloudflare?

### ISSUE THREAD

**Issue by norayr** [halogenica/beautifulhugo#189]
Hello, my [site](http://norayr.am/weblog) is being opened very slowly, and I see that the browser makes request to cdnjs.cloudflare.com, and those requests are very slowly from my connection.

Does that mean, that the beautifulhugo theme won't work without third party services?

Is it possible to download some of js and keep it locally?

What does cloudflare js? Can I just turn it off?

I'd like my site to work even if my country get cut off from Internet, which happens sometimes, because we only have one cable for the whole country. Thank you.

**Comment by NuLL3rr0r** [comment]
Yes, that's possible with some modification. If you don't want to use cloudflare, you can self-host those files and override the corresponding files inside a layout folder inside your blog's source.

I've already done that. Take a look at https://github.com/NuLL3rr0r/babaei-net-blog-source/tree/master/babaei.net/layouts

**Comment by halogenica** [comment]
I would love to make an option in the theme to be 100% self hosted. Would anyone (maybe @NuLL3rr0r ) be willing to submit a PR for this? I've tried to do this before and I was unable to figure out how to get an offline version of the google fonts used in this theme.

**Comment by NuLL3rr0r** [comment]
I would love to, but unfortunately, at the moment, I'm on a long vacation with no access to a proper PC or a good internet connection. For the google web fonts I always use this awesome tool https://github.com/majodev/google-webfonts-helper to download an offline version of any google web font. If there's no one to pick this up, I could do this when I'm back. But it probably won't happen in a month or two.

**Comment by iootaa** [comment]
This is now easy to solve: set `selfHosted` to `true\ inside config.toml:
`
[Params]
  selfHosted = true
`
see #245

**Comment by henryiii** [comment]
selfHosted = true fully supported
